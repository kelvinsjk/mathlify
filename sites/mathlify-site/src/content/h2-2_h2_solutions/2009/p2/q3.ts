import { Answer } from '$lib/classes/answer';
import { fnTerm } from 'mathlify';
import { mathlifierDj as mathlifier } from 'mathlifier';
import {
	generateAns,
	generateFracFnDefinition
} from '$content/h2-1_h2_learn/01_functions/04_more/01_self-inverse/02_practice';
import { EquationWorking, solve } from 'mathlify/working';
import { QED } from '$lib/typesetting/utils';

export const answer = new Answer();

// f(x) = ax/(bx-a)
// g(x) = 1/x

const fState: Parameters<typeof generateFracFnDefinition>[0] = {
	// (bx+k)/(cx+a)
	fnType: 'frac',
	//a: '-a',
	b: 'a',
	c: 'b',
	k: 0,
	n: 2,
	show: false
};
const fExp = generateFracFnDefinition(fState, { for: true })[1];
// a
{
	const { ans, soln } = generateAns(fState, fExp, 'f', { qed: true });
	answer.addPart(
		ans +
			mathlifier`\\
${{}}R_{f^2} = \\left( -\\infty, \\frac{a}{b} \\right) \\cup \\left( \\frac{a}{b}, \\infty \\right).`,
		soln +
			mathlifier`\n
$${{}} R_{f^2} = \\left( -\\infty, \\frac{a}{b} \\right) \\cup \\left( \\frac{a}{b}, \\infty \\right) \\; ${QED}`
	);
}
// b
{
	const soln = mathlifier`$${'align*'} R_g &= \\left( -\\infty, 0 \\right) \\cup \\left( 0, \\infty \\right)
\\\\ D_f &= \\left( -\\infty, \\frac{a}{b} \\right) \\cup \\left( \\frac{a}{b}, \\infty \\right)

Since ${'a'}
and ${'b'}
are non-zero,

$${{}} R_g \\not \\subseteq D_f

so the composite function ${'fg'}
does not exist ${QED}`;
	answer.addPart(
		mathlifier`The composite function ${'fg'}
does not exist as ${{}} {R_g \\not \\subseteq D_f.}`,
		soln
	);
}
// c
{
	const working = new EquationWorking(fnTerm('f^{-1}', 'x'), 'x');
	working.subIn({ 'f^{-1}': fExp });
	working.crossMultiply({ hide: true }).expand().swapSides({ hide: true });
	working.makeRhsZero().factorize.commonFactor();
	const { roots, working: rootsWorking } = solve.zeroProduct(working.eqn, 'x', { qed: true });
	const soln = mathlifier`$${'gather*'} ${working}

$${'align*'} ${rootsWorking}	`;
	const ans = mathlifier`${'x'} = ${roots.join(' \\text{ or }')}.`;
	answer.addPart(ans, soln);
}
