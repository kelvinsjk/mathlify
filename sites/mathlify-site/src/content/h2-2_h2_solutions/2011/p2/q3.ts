import { Answer } from '$lib/classes/answer';
import { mathlifierDj as mathlifier } from 'mathlifier';
import { Polynomial, sum } from 'mathlify';
import { logInverse } from '$content/h2-1_h2_learn/01_functions/02_inverse/03_formula/02_practice';
import { inverseRelationshipSolver } from '$content/h2-1_h2_learn/01_functions/02_inverse/05_relationship/02_practice-1';
import svg from '$static/images/h2/fns/2011p2q3b.svg?raw';
import { logTerm } from 'mathlify/fns';

export const answer = new Answer();

// ! Complete
// ln(2x+1) + 3

const argument = new Polynomial([2, 1]);
const b = 3;
const yMinusB = sum('y', -b);
const f = sum(logTerm(argument), b);

// a
{
	const { ans, soln } = logInverse(f, argument, yMinusB);
	answer.addPart(ans, soln);
}
// b
{
	const soln = `\`\`\` =html\n${svg}\n\`\`\`\n\n`;
	const ans = mathlifier`[Sketch](#soln-2).`;
	answer.addPart(ans, soln);
}
// c
const state = {
	fnType: 'log',
	a: 1,
	c: 2,
	b: 3
} as const;
{
	const { ans, soln } = inverseRelationshipSolver(state, f, {
		precision: 4,
		tys2011ExtraStep: true
	});
	answer.addPart(ans, soln);
}
