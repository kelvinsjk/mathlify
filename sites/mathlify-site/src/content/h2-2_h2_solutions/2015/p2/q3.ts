import { Answer } from '$lib/classes/answer';
import { Expression, Polynomial } from 'mathlify';
import { specialExistence } from '$content/h2-1_h2_learn/01_functions/02_inverse/01_existence/02_practice';
import { specialInverse } from '$content/h2-1_h2_learn/01_functions/02_inverse/03_formula/02_practice';
import { discriminantRange } from '$content/h2-1_h2_learn/01_functions/04_more/04_discriminant/02_practice';
import svg from '$static/images/h2/fns/2015p2q3a.svg?raw';
import { QED } from '$lib/typesetting/utils';

export const answer = new Answer();

// (a)
// | 1 / 1-x^2 |
// equivalent to | ba^2 / x^2 - a^2 |
// from the learn/practice section
// where b=-1 and a=1

// (b)
// (2+x) / (1-x^2)
// equivalent to (dx+e)/(c^2 - a^2x^2)

const [a, b] = [1, -1];
const exp = new Expression([a, '/', new Polynomial([a * a, 0, -1], { ascending: true })]);
const state: Parameters<typeof specialInverse>[0] = {
	fnType: 'special',
	a,
	b,
	restriction: { type: 'right', x: a, inclusive: false },
	definition: false,
	// dummies
	c: 1,
	unknownConstants: false
};

// ai
{
	const { ans, soln } = specialExistence(state);
	const imag = `\`\`\` =html\n${svg}\n\`\`\`\n\n`;
	answer.addSubPart(ans, imag + soln + ` $${QED}$`);
}
// aii
{
	const { ans, soln } = specialInverse(state, exp);
	answer.addSubPart(ans, soln);
}

// b
const stateB: Parameters<typeof discriminantRange>[0] = {
	a: 1,
	b: 0,
	c: 1,
	d: 1,
	e: 2,
	type: 3
};
{
	const { ans, soln } = discriminantRange(stateB);
	answer.addPart(ans, soln);
}
