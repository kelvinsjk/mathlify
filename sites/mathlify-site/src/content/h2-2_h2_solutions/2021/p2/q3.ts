import { Answer } from '$lib/classes/answer';
import { mathlifierDj as mathlifier } from 'mathlifier';
import { Expression, fnTerm, Polynomial, quotient } from 'mathlify';
import { fractionalInverseWorking } from '$content/h2-1_h2_learn/01_functions/02_inverse/03_formula/02_practice';
import { EquationWorking, ExpressionWorking } from 'mathlify/working';
import { QED } from '$lib/typesetting/utils';

export const answer = new Answer();

// ! Complete, but can be further abstracted

// 1/2 x^2 + 3
// (x+1)/(5x-1)
// (x+a)(2x+b)

const h = new Polynomial([[1, '/', 2], 0, 3]);
const g = quotient(new Polynomial([1, 1]), new Polynomial([5, -1]));
const verbatim = true;
const num = new Polynomial([1, 'a']);
const den = new Polynomial([2, 'b']);
const f = quotient(num, den);

// ai
{
	const x = 2;
	const h2 = fnTerm('h', x);
	const gh2 = fnTerm('g', h2);
	const working = new ExpressionWorking(gh2);
	working.subIn({ h }, { verbatim }).simplify().subIn({ g }, { verbatim }).simplify();
	const finalAns = working.expression;

	const soln = mathlifier`$${'align*'} ${working}`;
	const ans = mathlifier`${{}} ${gh2} = ${finalAns}.`;
	answer.addSubPart(ans, soln);
}
// aii
{
	const b = quotient(14, 10);
	const working = new EquationWorking(g, b);
	working.crossMultiply().expand().solve.linear('x');
	const x = working.eqn.rhs;
	const soln = mathlifier`$${'gather*'} g(x) = ${b.valueOf()} \\\\ ${working}`;
	const ans = mathlifier`${{}} x = ${x}.`;
	answer.addSubPart(ans, soln);
}
answer.newPart();
// bi
{
	const working = new EquationWorking(den, 0);
	const { root: k } = working.solve.linear('x');
	const soln = mathlifier`Note that division by zero is undefined
$${'gather*'}${working}

Hence ${{}}k=${k}
and it must be excluded from the domain of ${'f'}
since ${'f'}
is not defined when ${{}} x = ${k}`;
	const ans = mathlifier`${{}} k = ${k}.
\\
It must be excluded as ${'f'}
is not defined when ${{}} {x = ${k}.}`;
	answer.addSubPart(ans, soln);
}
// bii
let fInv: Expression;
{
	const { working, exp: fInv1 } = fractionalInverseWorking(f, false, { reportInverse: true });
	const b = -1;
	const a = quotient(-1, 2);
	fInv = f.subIn({ b });
	const soln =
		working +
		mathlifier`$${'align*'} f(x) &= f^{-1}(x) \\\\ ${f} &= ${fInv1} 

Since this is valid for all ${'x'}
in the domain of ${'f'},
by comparing,

$${{}} b = ${b} ${QED}

Substituting this into ${'f(x)'}:

$${{}} f(x) = ${fInv}

We observe that if ${{}} {a = ${a},}
then ${{}} {f(x) = ${fInv.subIn({ a })} = \\frac{1}{2}}
which is not a one-to-one function and does not have an inverse.

Hence ${'a'}
can take all real values except ${a}

$${{}} a \\in \\mathbb{R}, a \\neq ${a} ${QED}`;
	const ans = mathlifier`${{}} b = ${b}.
\\
${{}} a \\in \\mathbb{R}, a \\neq ${a}.`;
	answer.addSubPart(ans, soln);
}
// bIII
{
	const x = -4;
	const fInvX = fnTerm('f^{-1}', x);
	const working = new ExpressionWorking(fInvX, { startOnFirstLine: true });
	working
		.subIn({ 'f^{-1}': fnTerm('f', x) })
		.subIn({ f: fInv }, { verbatim })
		.simplify()
		.expandNegativeIntoQuotient();
	const finalAns = working.expression;
	const ans = mathlifier`${{}} f^{-1}(${x}) = ${finalAns}.`;
	const soln = mathlifier`$${'align*'} ${working}`;
	answer.addSubPart(ans, soln);
}
