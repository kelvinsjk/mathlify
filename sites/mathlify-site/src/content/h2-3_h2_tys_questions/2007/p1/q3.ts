/* The question here is the copyright of 
- Singapore Examinations and Assessment Board
- Cambridge Assessment International Education
- Ministry of Education, Singapore
- Government of Singapore

We provide this question on Mathlify under fair use for archival purposes.
These are questions more than 10 years old and are difficult to obtain
commercially.

For the newer questions, please purchase a copy of the
"ten year series" from authorized publishers and distributors.
*/

import { Expression, i, sum } from 'mathlify';
import { Question } from '$lib/classes/question';
import { mathlifierDj as mathlifier } from 'mathlifier';

export const question = new Question();

// ww^* + 2w = 3 + 4i

const w = new Expression('w');
const wStar = new Expression('w^*');
const lhs = sum([w, wStar], [2, w]);
const rhs = sum(3, [4, i]);
const aPlusBI = sum('a', [i, 'b']);

// a
{
	question.addPart(mathlifier`Out of syllabus (complex locus).`, { marks: 3 });
}
// b
{
	question.addPart(
		mathlifier`The complex number ${w}
is such that

$${lhs} = ${rhs},

where ${wStar}
is the complex conjugate of ${w}.
Find ${w}
in the form ${aPlusBI},
where ${'a'}
and ${'b'}
are real.`,
		{ marks: 4 }
	);
}
