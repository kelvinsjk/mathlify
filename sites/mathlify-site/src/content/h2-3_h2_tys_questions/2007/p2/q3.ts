/* The question here is the copyright of 
- Singapore Examinations and Assessment Board
- Cambridge Assessment International Education
- Ministry of Education, Singapore
- Government of Singapore

We provide this question on Mathlify under fair use for archival purposes.
These are questions more than 10 years old and are difficult to obtain
commercially.

For the newer questions, please purchase a copy of the
"ten year series" from authorized publishers and distributors.
*/

import { Polynomial } from 'mathlify';
import { Question } from '$lib/classes/question';
import { mathlifierDj as mathlifier } from 'mathlifier';

export const question = new Question();

const ascending = true;
const poly1 = new Polynomial([1, 1], { ascending });
const poly2 = new Polynomial([4, -1], { ascending });
const poly3 = new Polynomial([1, 0, 2], { ascending });

const exp1 = `{(${poly1})^n,}`;
const exp2 = `{(${poly2})^{\\frac{3}{2}} (${poly3})^{\\frac{3}{2}}}`;

// a
question.addPart(
	mathlifier`By successively differentiating ${exp1}
find Maclaurin's series for ${exp1}
up to and including the term in ${'x^3'}.`,
	{ marks: 4 }
);
// b
question.addPart(
	mathlifier`Obtain the expansion of ${exp2}
up to and including the term in ${'x^3'}.`,
	{ marks: 5 }
);
// c
question.addPart(
	mathlifier`Find the set of values of ${'x'}
for which the expansion in part (b) is valid.`,
	{ marks: 2 }
);
