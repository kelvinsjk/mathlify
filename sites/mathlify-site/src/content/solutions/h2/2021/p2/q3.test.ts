import { expect, it } from 'vitest';
import { answer } from './q3.soln';

it('2021p2q3', () => {
	expect(answer).toMatchSnapshot();
});
