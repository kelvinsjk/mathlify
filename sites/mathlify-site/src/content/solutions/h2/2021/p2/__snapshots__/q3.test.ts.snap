// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`2021p2q3 1`] = `
Answer {
  "answer": {
    "parts": [
      {
        "parts": [
          {
            "body": "<p><math><mrow><mi>g</mi><mi>h</mi><mrow><mo fence="true" form="prefix">(</mo><mn>2</mn><mo fence="true" form="postfix">)</mo></mrow><mo>=</mo></mrow><mrow><mfrac><mn>1</mn><mn>4</mn></mfrac><mi>.</mi></mrow></math></p>
",
          },
          {
            "body": "<p><math><mrow><mi>x</mi><mo>=</mo></mrow><mrow><mfrac><mn>2</mn><mn>5</mn></mfrac><mi>.</mi></mrow></math></p>
",
          },
        ],
      },
      {
        "parts": [
          {
            "body": "<p><math><mrow><mi>k</mi><mo>=</mo></mrow><mrow><mo form="prefix" stretchy="false">−</mo></mrow><mrow><mfrac><mi>b</mi><mn>2</mn></mfrac><mi>.</mi></mrow></math>
<br>
It must be excluded as <math><mi>f</mi></math>
is not defined when <math><mrow><mi>x</mi><mo>=</mo><mo form="prefix" stretchy="false">−</mo><mfrac><mi>b</mi><mn>2</mn></mfrac></mrow></math></p>
",
          },
          {
            "body": "<p><math><mrow><mi>b</mi><mo>=</mo></mrow><mrow><mo form="prefix" stretchy="false">−</mo></mrow><mrow><mn>1</mn><mi>.</mi></mrow></math>
<br>
<math><mrow><mi>a</mi><mo>∈</mo></mrow><mrow><mi>ℝ</mi><mo separator="true">,</mo></mrow><mrow><mi>a</mi><mo>≠</mo></mrow><mrow><mo form="prefix" stretchy="false">−</mo></mrow><mrow><mfrac><mn>1</mn><mn>2</mn></mfrac><mi>.</mi></mrow></math></p>
",
          },
          {
            "body": "<p><math><mrow><msup><mi>f</mi><mrow><mo lspace="0em" rspace="0em">−</mo><mn>1</mn></mrow></msup><mo form="prefix" stretchy="false">(</mo><mo form="prefix" stretchy="false">−</mo><mn>4</mn><mo form="postfix" stretchy="false">)</mo><mo>=</mo></mrow><mrow><mfrac><mrow><mn>4</mn><mo>−</mo><mi>a</mi></mrow><mn>9</mn></mfrac><mi>.</mi></mrow></math></p>
",
          },
        ],
      },
    ],
  },
  "solution": {
    "parts": [
      {
        "parts": [
          {
            "body": "<p><math display="block" class="tml-display" style="display:block math;"><mtable displaystyle="true" columnalign="right left"><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mrow></mrow></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mi>g</mi><mi>h</mi><mrow><mo fence="true" form="prefix">(</mo><mn>2</mn><mo fence="true" form="postfix">)</mo></mrow></mrow></mtd></mtr><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mrow></mrow></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><mi>g</mi><mrow><mo fence="true" form="prefix">(</mo><mfrac><mn>1</mn><mn>2</mn></mfrac><mrow><mo fence="true" form="prefix">(</mo><msup><mn>2</mn><mn>2</mn></msup><mo fence="true" form="postfix">)</mo></mrow><mo>+</mo><mn>3</mn><mo fence="true" form="postfix">)</mo></mrow></mrow></mtd></mtr><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mrow></mrow></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><mi>g</mi><mrow><mo fence="true" form="prefix">(</mo><mn>5</mn><mo fence="true" form="postfix">)</mo></mrow></mrow></mtd></mtr><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mrow></mrow></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><mfrac><mrow><mn>5</mn><mo>+</mo><mn>1</mn></mrow><mrow><mn>5</mn><mrow><mo fence="true" form="prefix">(</mo><mn>5</mn><mo fence="true" form="postfix">)</mo></mrow><mo>−</mo><mn>1</mn></mrow></mfrac></mrow></mtd></mtr><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mrow></mrow></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><mfrac><mn>1</mn><mn>4</mn></mfrac><mspace width="0.2778em"></mspace><mi>■</mi></mrow></mtd></mtr></mtable></math></p>
",
          },
          {
            "body": "<p><math display="block" class="tml-display" style="display:block math;"><mtable displaystyle="true"><mtr><mtd style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mi>g</mi><mo form="prefix" stretchy="false">(</mo><mi>x</mi><mo form="postfix" stretchy="false">)</mo><mo>=</mo><mn>1.4</mn></mrow></mtd></mtr><mtr><mtd style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mfrac><mrow><mi>x</mi><mo>+</mo><mn>1</mn></mrow><mrow><mn>5</mn><mi>x</mi><mo>−</mo><mn>1</mn></mrow></mfrac><mo>=</mo><mfrac><mn>7</mn><mn>5</mn></mfrac></mrow></mtd></mtr><mtr><mtd style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mn>5</mn><mrow><mo fence="true" form="prefix">(</mo><mi>x</mi><mo>+</mo><mn>1</mn><mo fence="true" form="postfix">)</mo></mrow><mo>=</mo><mn>7</mn><mrow><mo fence="true" form="prefix">(</mo><mn>5</mn><mi>x</mi><mo>−</mo><mn>1</mn><mo fence="true" form="postfix">)</mo></mrow></mrow></mtd></mtr><mtr><mtd style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mn>5</mn><mi>x</mi><mo>+</mo><mn>5</mn><mo>=</mo><mn>35</mn><mi>x</mi><mo>−</mo><mn>7</mn></mrow></mtd></mtr><mtr><mtd style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>−</mo><mn>30</mn><mi>x</mi><mo>=</mo><mo form="prefix" stretchy="false">−</mo><mn>12</mn></mrow></mtd></mtr><mtr><mtd style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mi>x</mi><mo>=</mo><mfrac><mn>2</mn><mn>5</mn></mfrac><mspace width="0.2778em"></mspace><mi>■</mi></mrow></mtd></mtr></mtable></math></p>
",
          },
        ],
      },
      {
        "parts": [
          {
            "body": "<p>Note that division by zero is undefined
<math display="block" class="tml-display" style="display:block math;"><mtable displaystyle="true"><mtr><mtd style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mn>2</mn><mi>x</mi><mo>+</mo><mi>b</mi><mo>=</mo><mn>0</mn></mrow></mtd></mtr><mtr><mtd style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mn>2</mn><mi>x</mi><mo>=</mo><mo form="prefix" stretchy="false">−</mo><mi>b</mi></mrow></mtd></mtr><mtr><mtd style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mi>x</mi><mo>=</mo><mo form="prefix" stretchy="false">−</mo><mfrac><mi>b</mi><mn>2</mn></mfrac></mrow></mtd></mtr></mtable></math></p>
<p>Hence <math><mrow><mi>k</mi><mo>=</mo></mrow><mrow><mo form="prefix" stretchy="false">−</mo></mrow><mrow><mfrac><mi>b</mi><mn>2</mn></mfrac></mrow></math>
and it must be excluded from the domain of <math><mi>f</mi></math>
since <math><mi>f</mi></math>
is not defined when <math><mrow><mi>x</mi><mo>=</mo></mrow><mrow><mo form="prefix" stretchy="false">−</mo></mrow><mrow><mfrac><mi>b</mi><mn>2</mn></mfrac><mspace width="0.2778em"></mspace><mi>■</mi></mrow></math></p>
",
          },
          {
            "body": "<p><math display="block" class="tml-display" style="display:block math;"><mtable displaystyle="true"><mtr><mtd style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mtext>Let </mtext><mi>y</mi><mo>=</mo><mfrac><mrow><mi>x</mi><mo>+</mo><mi>a</mi></mrow><mrow><mn>2</mn><mi>x</mi><mo>+</mo><mi>b</mi></mrow></mfrac></mrow></mtd></mtr><mtr><mtd style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mi>y</mi><mrow><mo fence="true" form="prefix">(</mo><mn>2</mn><mi>x</mi><mo>+</mo><mi>b</mi><mo fence="true" form="postfix">)</mo></mrow><mo>=</mo><mi>x</mi><mo>+</mo><mi>a</mi></mrow></mtd></mtr><mtr><mtd style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mn>2</mn><mi>y</mi><mi>x</mi><mo>+</mo><mi>y</mi><mi>b</mi><mo>=</mo><mi>x</mi><mo>+</mo><mi>a</mi></mrow></mtd></mtr><mtr><mtd style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mn>2</mn><mi>y</mi><mi>x</mi><mo>−</mo><mi>x</mi><mo>=</mo><mi>a</mi><mo>−</mo><mi>y</mi><mi>b</mi></mrow></mtd></mtr><mtr><mtd style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mi>x</mi><mrow><mo fence="true" form="prefix">(</mo><mn>2</mn><mi>y</mi><mo>−</mo><mn>1</mn><mo fence="true" form="postfix">)</mo></mrow><mo>=</mo><mi>a</mi><mo>−</mo><mi>y</mi><mi>b</mi></mrow></mtd></mtr><mtr><mtd style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mi>x</mi><mo>=</mo><mfrac><mrow><mi>a</mi><mo>−</mo><mi>y</mi><mi>b</mi></mrow><mrow><mn>2</mn><mi>y</mi><mo>−</mo><mn>1</mn></mrow></mfrac></mrow></mtd></mtr><mtr><mtd style="padding:0.7ex 0em 0.7ex 0em;"><mrow><msup><mi>f</mi><mrow><mo lspace="0em" rspace="0em">−</mo><mn>1</mn></mrow></msup><mo form="prefix" stretchy="false">(</mo><mi>x</mi><mo form="postfix" stretchy="false">)</mo><mo>=</mo><mfrac><mrow><mi>a</mi><mo>−</mo><mi>x</mi><mi>b</mi></mrow><mrow><mn>2</mn><mi>x</mi><mo>−</mo><mn>1</mn></mrow></mfrac></mrow></mtd></mtr></mtable></math></p>
<p><math display="block" class="tml-display" style="display:block math;"><mtable displaystyle="true" columnalign="right left"><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mi>f</mi><mo form="prefix" stretchy="false">(</mo><mi>x</mi><mo form="postfix" stretchy="false">)</mo></mrow></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><msup><mi>f</mi><mrow><mo form="prefix" stretchy="false" lspace="0em" rspace="0em">−</mo><mn>1</mn></mrow></msup><mo form="prefix" stretchy="false">(</mo><mi>x</mi><mo form="postfix" stretchy="false">)</mo></mrow></mtd></mtr><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mfrac><mrow><mi>x</mi><mo>+</mo><mi>a</mi></mrow><mrow><mn>2</mn><mi>x</mi><mo>+</mo><mi>b</mi></mrow></mfrac></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><mfrac><mrow><mi>a</mi><mo>−</mo><mi>x</mi><mi>b</mi></mrow><mrow><mn>2</mn><mi>x</mi><mo>−</mo><mn>1</mn></mrow></mfrac></mrow></mtd></mtr></mtable></math></p>
<p>Since this is valid for all <math><mi>x</mi></math>
in the domain of <math><mrow><mi>f</mi><mo separator="true">,</mo></mrow></math>
by comparing,</p>
<p><math display="block" class="tml-display" style="display:block math;"><mrow><mi>b</mi><mo>=</mo><mo form="prefix" stretchy="false">−</mo><mn>1</mn><mspace width="0.2778em"></mspace><mi>∎</mi></mrow></math></p>
<p>Substituting this into <math><mrow><mi>f</mi><mo form="prefix" stretchy="false">(</mo><mi>x</mi><mo form="postfix" stretchy="false">)</mo><mo lspace="0.2222em" rspace="0.2222em">:</mo></mrow></math></p>
<p><math display="block" class="tml-display" style="display:block math;"><mrow><mi>f</mi><mo form="prefix" stretchy="false">(</mo><mi>x</mi><mo form="postfix" stretchy="false">)</mo><mo>=</mo><mfrac><mrow><mi>x</mi><mo>+</mo><mi>a</mi></mrow><mrow><mn>2</mn><mi>x</mi><mo>−</mo><mn>1</mn></mrow></mfrac></mrow></math></p>
<p>We observe that if <math><mrow><mi>a</mi><mo>=</mo><mo form="prefix" stretchy="false">−</mo><mfrac><mn>1</mn><mn>2</mn></mfrac><mo separator="true" lspace="0em" rspace="0em">,</mo></mrow></math>
then <math><mrow><mi>f</mi><mo form="prefix" stretchy="false">(</mo><mi>x</mi><mo form="postfix" stretchy="false">)</mo><mo>=</mo><mfrac><mrow><mi>x</mi><mo>−</mo><mfrac><mn>1</mn><mn>2</mn></mfrac></mrow><mrow><mn>2</mn><mi>x</mi><mo>−</mo><mn>1</mn></mrow></mfrac><mo>=</mo><mfrac><mn>1</mn><mn>2</mn></mfrac></mrow></math>
which is not a one-to-one function and does not have an inverse.</p>
<p>Hence <math><mi>a</mi></math>
can take all real values except <math><mrow><mo>−</mo></mrow><mrow><mfrac><mn>1</mn><mn>2</mn></mfrac></mrow></math></p>
<p><math display="block" class="tml-display" style="display:block math;"><mrow><mi>a</mi><mo>∈</mo><mi>ℝ</mi><mo separator="true">,</mo><mi>a</mi><mo>≠</mo><mo form="prefix" stretchy="false">−</mo><mfrac><mn>1</mn><mn>2</mn></mfrac><mspace width="0.2778em"></mspace><mi>∎</mi></mrow></math></p>
",
          },
          {
            "body": "<p><math display="block" class="tml-display" style="display:block math;"><mtable displaystyle="true" columnalign="right left"><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><msup><mi>f</mi><mrow><mo lspace="0em" rspace="0em">−</mo><mn>1</mn></mrow></msup><mrow><mo fence="true" form="prefix">(</mo><mo>−</mo><mn>4</mn><mo fence="true" form="postfix">)</mo></mrow></mrow></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><mi>f</mi><mrow><mo fence="true" form="prefix">(</mo><mo>−</mo><mn>4</mn><mo fence="true" form="postfix">)</mo></mrow></mrow></mtd></mtr><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mrow></mrow></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><mfrac><mrow><mo form="prefix" stretchy="false" lspace="0em" rspace="0em">−</mo><mn>4</mn><mo>+</mo><mi>a</mi></mrow><mrow><mn>2</mn><mrow><mo fence="true" form="prefix">(</mo><mo>−</mo><mn>4</mn><mo fence="true" form="postfix">)</mo></mrow><mo>−</mo><mn>1</mn></mrow></mfrac></mrow></mtd></mtr><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mrow></mrow></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><mo form="prefix" stretchy="false">−</mo><mfrac><mrow><mo form="prefix" stretchy="false" lspace="0em" rspace="0em">−</mo><mn>4</mn><mo>+</mo><mi>a</mi></mrow><mn>9</mn></mfrac></mrow></mtd></mtr><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mrow></mrow></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><mfrac><mrow><mn>4</mn><mo>−</mo><mi>a</mi></mrow><mn>9</mn></mfrac><mspace width="0.2778em"></mspace><mi>■</mi></mrow></mtd></mtr></mtable></math></p>
",
          },
        ],
      },
    ],
  },
}
`;
